tasks/object-detection-yolov8/src/predict.py

```diff
@@ -13,27 +13,42 @@ def classByLabelId(labelId: int, classes: ImageDatasetClasses) -> Optional[Image
     return classes.classByLabel(classes.labels[labelId])
 
 
-def run(model: YOLO, dataset: ComputerVisionDataset, resultPath: Path) -> None:
-    for sample in dataset.samples:
-        result: Results = model.predict(sample.imagePath, save = True, project = "./results")[0]
-        plt.imshow(result.orig_img)
-
-        if result.boxes is not None:
-            for minX, minY, maxX, maxY, confidence, labelId in result.boxes.data:
-                box = BBox.create(minX, minY, maxX, maxY)
-
-                clazz = classByLabelId(int(labelId), dataset.classes)
-                if clazz is None:
-                    continue
-
-                plt.gca().add_patch(pth.Rectangle(
-                    (float(box.minX), float(box.minY)),
-                    float(box.width),
-                    float(box.height),
-                    linewidth = 3,
-                    edgecolor = clazz.color,
-                    facecolor = "none"
-                ))
-
-        plt.savefig(resultPath / f"{sample.id}.png")
-        plt.close()
+def processResult(result: Results, classes: list[ImageDatasetClasses], savePath: Path):
+    fig = plt.figure(num = 1, clear = True)
+    plt.imshow(result.orig_img)
+
+    if result.boxes is not None:
+        for minX, minY, maxX, maxY, confidence, labelId in result.boxes.data:
+            box = BBox.create(minX, minY, maxX, maxY)
+
+            clazz = classByLabelId(int(labelId), classes)
+            if clazz is None:
+                continue
+
+            fig.gca().add_patch(pth.Rectangle(
+                (float(box.minX), float(box.minY)),
+                float(box.width),
+                float(box.height),
+                linewidth = 3,
+                edgecolor = clazz.color,
+                facecolor = "none"
+            ))
+
+    plt.savefig(savePath)
+
+
+def run(model: YOLO, dataset: ComputerVisionDataset, resultPath: Path, batchSize: int) -> None:
+    sampleBatch: list[Path] = []
+    for i, sample in enumerate(dataset.samples):
```

Comment:
Isn't there a better way to do batching? ex. `for i in range(0, len(dataset.samples), batchSize)`, and then for the remained it is possible to get it like `dataset.count % batchSize`.